probe-rs/src/probe/ti_icdi/mod.rs

```diff
@@ -0,0 +1,289 @@
+        self.device
+            .send_remote_command(b"debug hreset")?
+            .check_cmd_result()
+    }
+
+    fn target_reset_assert(&mut self) -> Result<(), DebugProbeError> {
+        self.device
+            .send_remote_command(b"debug sreset")
+            .and_then(|r| r.check_cmd_result())
+    }
+
+    fn target_reset_deassert(&mut self) -> Result<(), DebugProbeError> {
+        self.device
+            .send_remote_command(b"debug hreset")
+            .and_then(|r| r.check_cmd_result())
+    }
+
+    fn select_protocol(&mut self, protocol: WireProtocol) -> Result<(), DebugProbeError> {
+        match protocol {
+            WireProtocol::Jtag => {
+                self.protocol = protocol;
+                Ok(())
+            }
+            _ => Err(DebugProbeError::UnsupportedProtocol(protocol)),
+        }
+    }
+
+    fn has_arm_interface(&self) -> bool {
+        true
+    }
+    fn try_get_arm_interface<'probe>(
+        self: Box<Self>,
+    ) -> Result<Box<dyn ArmProbeInterface + 'probe>, (Box<dyn DebugProbe>, DebugProbeError)> {
+        Ok(self)
+    }
+
+    fn into_probe(self: Box<Self>) -> Box<dyn DebugProbe> {
+        self
+    }
+}
+
+impl ArmProbeInterface for IcdiProbe {
+    fn memory_interface(&mut self, access_port: MemoryAp) -> Result<Memory<'_>, Error> {
+        Ok(Memory::new(self, access_port))
+    }
+
+    fn ap_information(&self, access_port: GenericAp) -> Option<&ApInformation> {
+        if access_port.port_number() != 0 {
+            return None;
+        }
+        Some(&ApInformation::MemoryAp(MemoryApInformation {
+            port_number: 0,
+            only_32bit_data_size: false,
+            debug_base_address: 0xE00FF000, // FIXME: This might only be true for Cortex-M4
+            supports_hnonsec: false,
+        }))
+    }
+
+    fn num_access_ports(&self) -> usize {
+        1
+    }
+
+    fn read_from_rom_table(&mut self) -> Result<Option<ArmChipInfo>, Error> {
+        let baseaddr = 0xE00FF000; // FIXME: This might only be true for Cortex-M4
+        let mut memory = self.as_memory();
+        let component = Component::try_parse(&mut memory, baseaddr)
+            .map_err(ProbeRsError::architecture_specific)?;
+
+        if let Component::Class1RomTable(component_id, _) = component {
+            if let Some(jep106) = component_id.peripheral_id().jep106() {
+                return Ok(Some(ArmChipInfo {
+                    manufacturer: jep106,
+                    part: component_id.peripheral_id().part(),
+                }));
+            }
+        }
+        Ok(None)
+    }
+
+    fn target_reset_deassert(&mut self) -> Result<(), Error> {
+        self.device
+            .send_remote_command(b"debug hreset")
+            .and_then(|response| response.check_cmd_result())
+            .map_err(Error::Probe)
+    }
+
+    fn close(self: Box<Self>) -> Probe {
+        Probe::new(*self)
+    }
+}
+
+impl DapAccess for IcdiProbe {
```

Comment:
It should be possible to not implement those traits (akin to ST-Link) and not just return `CommandNotSupportedByProbe`.